src/main/java/com/linkedin/kmf/services/MultiClusterTopicManagementService.java

```diff
@@ -91,18 +92,19 @@ public MultiClusterTopicManagementService(Map<String, Object> props, String serv
     _scheduleIntervalMs = config.getInt(MultiClusterTopicManagementServiceConfig.REBALANCE_INTERVAL_MS_CONFIG);
     _preferredLeaderElectionIntervalMs = config.getLong(MultiClusterTopicManagementServiceConfig.PREFERRED_LEADER_ELECTION_CHECK_INTERVAL_MS_CONFIG);
     _completableFuture = new CompletableFuture<>();
-    _executor = Executors.newSingleThreadScheduledExecutor(new ThreadFactory() {
-      @Override
-      public Thread newThread(Runnable r) {
-        return new Thread(r, _serviceName + "-multi-cluster-topic-management-service");
-      }
-    });
+    _executor = Executors.newSingleThreadScheduledExecutor(
+      r -> new Thread(r, _serviceName + "-multi-cluster-topic-management-service"));
+    _topicPartitionReady.complete(null);
   }
 
   public CompletableFuture<Void> topicManagementReady() {
```

Comment:
Perhaps this method should initiate the async operation and return the future.

Reply:
Will put in the backlog the Asynchronous operation for `topicmanagementReady()`.